codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -109,12 +124,23 @@ public void incrementWindowSize(ChannelHandlerContext ctx, Http2Stream stream, i
     @Override
     public void consumeBytes(ChannelHandlerContext ctx, Http2Stream stream, int numBytes)
             throws Http2Exception {
-        state(stream).consumeBytes(ctx, numBytes);
+        if (stream.id() == CONNECTION_STREAM_ID) {
+            throw new UnsupportedOperationException("Returning bytes for the connection window is not supported");
+        }
+        if (numBytes <= 0) {
+            throw new IllegalArgumentException("numBytes must be positive");
+        }
+
+        // Streams automatically consume all remaining bytes when they are closed, so just ignore
+        // if already closed.
+        if (!isClosed(stream)) {
+            state(stream).consumeBytes(ctx, numBytes);
+        }
     }
 
     @Override
     public int unconsumedBytes(Http2Stream stream) {
-        return state(stream).unconsumedBytes();
+        return isClosed(stream) ? 0 : state(stream).unconsumedBytes();
```

Comment:
Do we have to do this?  If we are forcing the unconsumed bytes to 0 in the connection listener, is that good enough...or is this method used while the stream is being closed before the listener is notified?